### DISCUSSION TITLE
Should I prefer using the .spec file or generating it via pyinstaller each time?

### DISCUSSION THREAD

**Discussion by AlexP11223** [pyinstaller/pyinstaller#8349] [PyInstaller]
Is there a reason to `.gitignore` the `.spec` file and call `pyinstaller` with all options instead? Does it generate the same file on all OSes if the same options are used?

**Comment by bwoodsend** [comment]
> Does it generate the same file on all OSes if the same options are used?

Almost. #8004 means that you get some backslash file separators which you'll need to swap for forward slashes if you generate on Windows. That is a defect. And `--windowed` mode on macOS is a bit different so if you were using that mode with a cross platform spec file, you'd likely need to wrap the `app = BUNDLE()` section in a `if platform.system() == "Darwin"` block.

> Should I prefer using the .spec file or generating it via pyinstaller each time?

It's up to you. The spec file lets you put custom build steps in, access some options that aren't available from the CLI, find resources to package dynamically, etc. Also, if you require many CLI flags, it's a lot more convenient to use `pyinstaller foo.spec` than it is to use `pyinstaller --lots --of --stuff --to --remember foo.py`. But if that's not functionality you need then by all means pretend the `.spec` doesn't exist.
